<launch>
  <machine name="robot" address="$(env ROBOT)" env-loader="/home/jfstepha/catkin_ws/env.sh" user="jfstepha" timeout="240"/>
  <!-- see http://www.ros.org/wiki/roslaunch/XML/machine -->

  <!-- change this next line to include the launch file for your robot's hardware drivers here -->
  
  <param name="robot_description" textfile="$(find yertle)/yertle.urdf" /> 
  
  <node machine="robot" name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" >
  <param name="publish_frequency" value="50" />
  </node>
  
  <node machine="robot" name="laptop_bat" pkg="yertle" type="laptop_bat.py" />
  <node machine="robot" name="laptop_wifi" pkg="yertle" type="laptop_wifi.py" >
     <param name="range" value="70" />
  </node>
  <node machine="robot" name="range0" pkg="yertle" type="int2range.py" output="screen" >
    <param name="frame" value="range0"/>
    <param name="rolling_pts" value="5" />
    <param name="max_range" value="0.30" />
    <remap from="range_int" to="/arduino/sensor/range0"/>
    <remap from="range" to="range_right"/>
    <remap from ="scan" to="scan_right"/>
  </node>
  <node machine="robot" name="range1" pkg="yertle" type="int2range.py" >
    <param name="frame" value="range1"/>
    <param name="rolling_pts" value="5" />
    <param name="max_range" value="0.30" />
    <remap from="range_int" to="/arduino/sensor/range1"/>
    <remap from="range" to="range_center"/>
    <remap from ="scan" to="scan_center"/>
  </node>
  <node machine="robot" name="range2" pkg="yertle" type="int2range.py" >
    <param name="frame" value="range2"/>
    <param name="rolling_pts" value="5" />
    <param name="max_range" value="0.30" />
    <remap from="range_int" to="/arduino/sensor/range2"/>
    <remap from="range" to="range_left"/>
    <remap from ="scan" to="scan_left"/>
  </node>
  <node machine="robot" name="arduino" pkg="ros_arduino_python" type="arduino_node.py" output="screen">
      <rosparam file="$(find yertle)/my_arduino_params.yaml" command="load" />
      <remap from="cmd_vel" to="twist"/>
  </node>

</launch>

